backend/stakepoold/server.go

```diff
@@ -459,60 +460,55 @@ func loadData(spd *stakepool.Stakepoold, dataKind string) error {
 		return errors.New("unhandled data kind of " + dataKind)
 	}
 
-	if fileExists(spd.DataPath) {
-		files, err := ioutil.ReadDir(spd.DataPath)
-		if err != nil {
-			return err
-		}
+	if !fileExists(spd.DataPath) {
+		return errors.New("loadData - path " + spd.DataPath + " does not exist")
+	}
 
-		var lastseen string
+	files, err := ioutil.ReadDir(spd.DataPath)
+	if err != nil {
+		return err
+	}
 
-		for i, file := range files {
-			log.Debugf("entry %d => %s", i, file.Name())
-			if strings.HasPrefix(file.Name(), strings.ToLower(dataKind)) &&
-				strings.Contains(file.Name(), dataVersion) &&
-				strings.HasSuffix(file.Name(), ".gob") {
-				lastseen = file.Name()
-			}
-		}
+	var lastseen string
 
-		// we could warn/error here but it's not really a problem.
-		// maybe the admin deleted the gob files to reset the cache
-		// or the cache hasn't been initialized yet.
-		if lastseen == "" {
-			return nil
+	for i, file := range files {
+		log.Debugf("entry %d => %s", i, file.Name())
+		if strings.HasPrefix(file.Name(), strings.ToLower(dataKind)) &&
+			strings.Contains(file.Name(), dataVersion) &&
+			strings.HasSuffix(file.Name(), ".gob") {
+			lastseen = file.Name()
 		}
+	}
 
-		fullPath := filepath.Join(spd.DataPath, lastseen)
-
-		r, err := os.Open(fullPath)
-		if err != nil {
-			return err
-		}
-		dec := gob.NewDecoder(r)
-		switch dataKind {
-		case "AddedLowFeeTickets":
-			err = dec.Decode(&spd.AddedLowFeeTicketsMSA)
-			if err != nil {
-				return err
-			}
-		case "LiveTickets":
-			err = dec.Decode(&spd.LiveTicketsMSA)
-			if err != nil {
-				return err
-			}
-		case "UserVotingConfig":
-			err = dec.Decode(&spd.UserVotingConfig)
-			if err != nil {
-				return err
-			}
-		}
-		log.Infof("Loaded %s from %s", dataKind, fullPath)
+	// we could warn/error here but it's not really a problem.
+	// maybe the admin deleted the gob files to reset the cache
+	// or the cache hasn't been initialized yet.
+	if lastseen == "" {
 		return nil
 	}
 
-	// shouldn't get here -- data dir is created on startup
-	return errors.New("loadData - path " + spd.DataPath + " does not exist")
+	fullPath := filepath.Join(spd.DataPath, lastseen)
+
+	r, err := os.Open(fullPath)
+	if err != nil {
+		return err
+	}
+
```

Comment:
I think that r (*os.File) should be closed at some point, although the godoc doesn't explicitly say so.

Reply:
Done